astra-sim/workload/Workload.cc

```diff
@@ -151,127 +155,145 @@ void Workload::issue_remote_mem(shared_ptr<Chakra::ETFeederNode> node) {
+  wlhd->node_id = node->id();
+  sys->remote_mem->issue(node->tensor_size(), wlhd);
 }
 
 void Workload::issue_comp(shared_ptr<Chakra::ETFeederNode> node) {
-  assert(node->getChakraNode()->node_type() == ChakraNodeType::COMP_NODE);
   hw_resource->occupy(node);
 
   if (sys->roofline_enabled) {
     WorkloadLayerHandlerData* wlhd = new WorkloadLayerHandlerData;
-    wlhd->node_id = node->getChakraNode()->id();
+    wlhd->node_id = node->id();
 
     double operational_intensity =
-        static_cast<double>(node->getChakraNode()->num_ops()) /
-        static_cast<double>(node->getChakraNode()->tensor_size());
+        static_cast<double>(node->num_ops()) /
+        static_cast<double>(node->tensor_size());
     double perf = sys->roofline->get_perf(operational_intensity);
     double elapsed_time =
-        static_cast<double>(node->getChakraNode()->num_ops()) / perf;
-    uint64_t simulated_run_time =
+        static_cast<double>(node->num_ops()) / perf;
+    uint64_t runtime =
         static_cast<uint64_t>(elapsed_time * static_cast<double>(FREQ));
-    sys->register_event(this, EventType::General, wlhd, simulated_run_time);
+    sys->register_event(this, EventType::General, wlhd, runtime);
   } else {
     WorkloadLayerHandlerData* wlhd = new WorkloadLayerHandlerData;
-    wlhd->node_id = node->getChakraNode()->id();
+    wlhd->node_id = node->id();
 
     sys->register_event(
         this,
         EventType::General,
         wlhd,
-        node->getChakraNode()->simulated_run_time());
+        // chakra runtimes are in microseconds and we should convert it into nanoseconds
+        node->runtime() * 1000);
   }
 }
 
 void Workload::issue_comm(shared_ptr<Chakra::ETFeederNode> node) {
   hw_resource->occupy(node);
 
   vector<bool> involved_dim;
-  for (int i = 0; i < node->getChakraNode()->involved_dim_size(); i++) {
-    involved_dim.push_back(node->getChakraNode()->involved_dim(i));
+  for (int i = 0; i < node->involved_dim_size(); i++) {
+    involved_dim.push_back(node->involved_dim(i));
   }
 
-  if (node->getChakraNode()->node_type() == ChakraNodeType::COMM_COLL_NODE) {
-    if (node->getChakraNode()->comm_type() ==
+  if (!node->is_cpu_op()
+      && (node->type() == ChakraNodeType::COMM_COLL_NODE)) {
+    if (node->comm_type() ==
         ChakraCollectiveCommType::ALL_REDUCE) {
       DataSet* fp = sys->generate_all_reduce(
-          node->getChakraNode()->comm_size(),
+          node->comm_size(),
           involved_dim,
           comm_group,
-          node->getChakraNode()->comm_priority());
-      collective_comm_node_id_map[fp->my_id] = node->getChakraNode()->id();
+          node->comm_priority());
+      collective_comm_node_id_map[fp->my_id] = node->id();
+      collective_comm_wrapper_map[fp->my_id] = fp;
       fp->set_notifier(this, EventType::CollectiveCommunicationFinished);
 
     } else if (
-        node->getChakraNode()->comm_type() ==
+        node->comm_type() ==
```

Comment:
What is going on with the coding style here?  I realize you are just following the prior coding style, but I don't understand why there is a new line here.  The first half of the condition would fit on the prior line.  Can you make this and the following "else if" style follow the pior style shown on line 201?

Reply:
You will understand this change when you take a look at another PR in Chakra: https://github.com/TaekyungHeo/chakra-1/pull/2. ASTRA-sim relies on et_feeder in Chakra for node definitions and the frontend. We've made updates to how we retrieve comm_type in that PR.

Reply:
I hope to get the time to review your other changes soon, however my suggestion was much simplier and just related to the coding style.  I simply wanted you to change:

"    } else if (
        node->comm_type() =="

to

"    } else if (node->comm_type() == "

That matches the style in the rest of the file.

Reply:
Updated!